manifests/versionlock.pp

```diff
@@ -12,12 +12,23 @@
 #   Wildcards may be used within token slots, but must not cover seperators,
 #   e.g., 'b*sh-4.1.2-9.*' covers Bash version 4.1.2, revision 9 on all
 #   architectures.
+#   On Suse systems below version 15 the resource title may only consist of the package name.
 #
 define zypprepo::versionlock {
   require zypprepo::plugin::versionlock
 
-  assert_type(Zypprepo::VersionlockString, $name) |$_expected, $actual | {
-    fail("Package name must be formatted as %{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}, not \'${actual}\'. See Zypprepo::Versionlock documentation for details.")
+  # Versionlock on SLES12 is different
+  case $facts['os']['release']['major'] {
+    '15': {
```

Comment:
should we flip the logic? This currently only matches for 15, but it probably valid for all new SLES releases?